### ISSUE TITLE
groupby loses names on distributed tables and errors with tuples

### ISSUE THREAD

**Issue by piever** [JuliaComputing/JuliaDB.jl#158]
Not sure why, but naming seems to fail on distributed tables, while it works

```julia
julia> t = table(1:10, 1:10, names = [:x, :y], chunks = 2)
Distributed Table with 10 rows in 2 chunks:
x   y
──────
1   1
2   2
3   3
4   4
5   5
6   6
7   7
8   8
9   9
10  10

julia> groupby(length, t, :x)
Distributed Table with 10 rows in 1 chunks:
1   2
─────
1   1
2   1
3   1
4   1
5   1
6   1
7   1
8   1
9   1
10  1

julia> groupby(length, collect(t), :x)
Table with 10 rows, 2 columns:
x   length
──────────
1   1
2   1
3   1
4   1
5   1
6   1
7   1
8   1
9   1
10  1
```

With explicit names it even errors:

```julia
julia> groupby(@NT(l = length), t, :x)
ERROR: syntax: duplicate field name: "x" is not unique
namedtuple(::Symbol, ::Vararg{Symbol,N} where N) at /Users/pietro/.julia/v0.6/IndexedTables/src/utils.jl:221
#convert#105(::Array{Any,1}, ::Function, ::Type{IndexedTables.NextTable}, ::IndexedTables.Columns{NamedTuples._NT_x{Int64},NamedTuples._NT_x{Array{Int64,1}}}, ::IndexedTables.Columns{NamedTuples._NT_x_y{Int64,Int64},NamedTuples._NT_x_y{Array{Int64,1},Array{Int64,1}}}) at /Users/pietro/.julia/v0.6/IndexedTables/src/table.jl:576
(::Base.#kw##convert)(::Array{Any,1}, ::Base.#convert, ::Type{IndexedTables.NextTable}, ::IndexedTables.Columns{NamedTuples._NT_x{Int64},NamedTuples._NT_x{Array{Int64,1}}}, ::IndexedTables.Columns{NamedTuples._NT_x_y{Int64,Int64},NamedTuples._NT_x_y{Array{Int64,1},Array{Int64,1}}}) at ./<missing>:0
#reindex#162(::Array{Any,1}, ::Function, ::Type{T} where T, ::IndexedTables.NextTable{IndexedTables.Columns{NamedTuples._NT_x_y{Int64,Int64},NamedTuples._NT_x_y{Array{Int64,1},Array{Int64,1}}}}, ::Tuple{Symbol}, ::Tuple{Symbol,Symbol}) at /Users/pietro/.julia/v0.6/IndexedTables/src/selection.jl:194
#reindex#162(::Array{Any,1}, ::Function, ::Type{T} where T, ::IndexedTables.NextTable{IndexedTables.Columns{NamedTuples._NT_x_y{Int64,Int64},NamedTuples._NT_x_y{Array{Int64,1},Array{Int64,1}}}}, ::Symbol, ::Tuple{Symbol,Symbol}) at /Users/pietro/.julia/v0.6/IndexedTables/src/selection.jl:185
#reindex#163(::Array{Any,1}, ::Function, ::IndexedTables.NextTable{IndexedTables.Columns{NamedTuples._NT_x_y{Int64,Int64},NamedTuples._NT_x_y{Array{Int64,1},Array{Int64,1}}}}, ::Symbol, ::Tuple{Symbol,Symbol}) at /Users/pietro/.julia/v0.6/IndexedTables/src/selection.jl:199
(::JuliaDB.#sortandsample#202{Symbol,Tuple{Symbol,Symbol}})(::IndexedTables.NextTable{IndexedTables.Columns{NamedTuples._NT_x_y{Int64,Int64},NamedTuples._NT_x_y{Array{Int64,1},Array{Int64,1}}}}, ::Int64, ::Bool) at /Users/pietro/.julia/v0.6/JuliaDB/src/sort.jl:58
do_task(::Dagger.Context, ::Dagger.OSProc, ::Int64, ::Function, ::Tuple{Dagger.Chunk{IndexedTables.NextTable{IndexedTables.Columns{NamedTuples._NT_x_y{Int64,Int64},NamedTuples._NT_x_y{Array{Int64,1},Array{Int64,1}}}},MemPool.DRef},Int64,Bool}, ::Bool, ::Bool, ::Bool) at /Users/pietro/.julia/v0.6/Dagger/src/compute.jl:319
(::Base.Distributed.##135#136{Dagger.#do_task,Tuple{Dagger.Context,Dagger.OSProc,Int64,JuliaDB.#sortandsample#202{Symbol,Tuple{Symbol,Symbol}},Tuple{Dagger.Chunk{IndexedTables.NextTable{IndexedTables.Columns{NamedTuples._NT_x_y{Int64,Int64},NamedTuples._NT_x_y{Array{Int64,1},Array{Int64,1}}}},MemPool.DRef},Int64,Bool},Bool,Bool,Bool},Array{Any,1}})() at ./distributed/remotecall.jl:314
run_work_thunk(::Base.Distributed.##135#136{Dagger.#do_task,Tuple{Dagger.Context,Dagger.OSProc,Int64,JuliaDB.#sortandsample#202{Symbol,Tuple{Symbol,Symbol}},Tuple{Dagger.Chunk{IndexedTables.NextTable{IndexedTables.Columns{NamedTuples._NT_x_y{Int64,Int64},NamedTuples._NT_x_y{Array{Int64,1},Array{Int64,1}}}},MemPool.DRef},Int64,Bool},Bool,Bool,Bool},Array{Any,1}}, ::Bool) at ./distributed/process_messages.jl:56
#remotecall_fetch#140(::Array{Any,1}, ::Function, ::Function, ::Base.Distributed.LocalProcess, ::Dagger.Context, ::Vararg{Any,N} where N) at ./distributed/remotecall.jl:339
remotecall_fetch(::Function, ::Base.Distributed.LocalProcess, ::Dagger.Context, ::Vararg{Any,N} where N) at ./distributed/remotecall.jl:339
#remotecall_fetch#144(::Array{Any,1}, ::Function, ::Function, ::Int64, ::Dagger.Context, ::Vararg{Any,N} where N) at ./distributed/remotecall.jl:367
macro expansion at /Users/pietro/.julia/v0.6/Dagger/src/compute.jl:332 [inlined]
(::Dagger.##69#70{Dagger.Context,Dagger.OSProc,Int64,JuliaDB.#sortandsample#202{Symbol,Tuple{Symbol,Symbol}},Tuple{Dagger.Chunk{IndexedTables.NextTable{IndexedTables.Columns{NamedTuples._NT_x_y{Int64,Int64},NamedTuples._NT_x_y{Array{Int64,1},Array{Int64,1}}}},MemPool.DRef},Int64,Bool},Channel{Any},Bool,Bool,Bool})() at ./event.jl:73

julia> groupby(@NT(l = length), collect(t), :x)
Table with 10 rows, 2 columns:
x   l
─────
1   1
2   1
3   1
4   1
5   1
6   1
7   1
8   1
9   1
10  1
```

**Comment by piever** [comment]
After looking at the code I think I see some issues (due to not updating JuliaDB when refactoring `groupby` in IndexedTables). `select=f isa Tup ? valuenames(t)` seems the culprit for the error, not sure what's cause issues with the renaming. One possibility is that it's messing up because `select isa Tuple` is a bad way of checking whether we're using a scalar or not because of special selectors, for example `Not(by) isa Tuple == false`

**Comment by piever** [comment]
Fixed by #163
